src/main/java/uk/gov/hmcts/divorce/common/service/task/ResetConditionalOrderFlags.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.divorce.common.service.task;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.task.CaseTask;
+
+import static uk.gov.hmcts.ccd.sdk.type.YesOrNo.NO;
+
+@Component
+@Slf4j
+public class ResetConditionalOrderFlags implements CaseTask {
+    @Override
+    public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
+        log.info("Resetting conditional order flags for case id {} ", caseDetails.getId());
+
+        final CaseData data = caseDetails.getData();
+
+        data.getConditionalOrder().getConditionalOrderApplicant1Questions().setIsSubmitted(NO);
+        data.getConditionalOrder().getConditionalOrderApplicant1Questions().setIsDrafted(NO);
```

Comment:
Would this throw NPE if it is a joint application and only applicant2 has applied for a conditional order and need a reset?

Reply:
@FaisalMoJ Thanks for checking, it should be okay because ConditionalOrderQuestions has a default value (see ConditionalOrder model file). In fact, we have been using this same code in caseworker NoC for a couple of years and my intention is to create a small tech debt for us to refactor the NoC events so they also use this case task